build.go

```diff
@@ -139,21 +143,57 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		return err
 	}
 
-	return c.lifecycle.Execute(ctx, build.LifecycleOptions{
+	lifecycleOpts := build.LifecycleOptions{
 		AppPath:            appPath,
 		Image:              imageRef,
 		Builder:            ephemeralBuilder,
 		RunImage:           runImageName,
 		ClearCache:         opts.ClearCache,
 		Publish:            opts.Publish,
+		TrustBuilder:       opts.TrustBuilder,
 		HTTPProxy:          proxyConfig.HTTPProxy,
 		HTTPSProxy:         proxyConfig.HTTPSProxy,
 		NoProxy:            proxyConfig.NoProxy,
 		Network:            opts.ContainerConfig.Network,
 		Volumes:            platformVolumes,
 		DefaultProcessType: opts.DefaultProcessType,
 		FileFilter:         opts.FileFilter,
-	})
+	}
```

Comment:
UseCreator and LifecycleImage aren't initialized here.
I understand that logically it works since you initialize each one of them in the relevant case below, but I think it will be more readable if they'll be initialized here and then the relevant changes will be made below.